xbmc/video/VideoInfoScanner.cpp

```diff
@@ -237,201 +246,159 @@ namespace VIDEO
     CServiceBroker::GetGUI()->GetWindowManager().SendThreadMessage(msg);
   }
 
-  bool CVideoInfoScanner::DoScan(const std::string& strDirectory)
+  void CVideoInfoScanner::loadRemoteItems(const std::string& strDirectory, CFileItemList& remoteDirectoryItems, SScanSettings rootPathScraperScanSettings, CONTENT_TYPE contentType_AsDefinedInScraper)
   {
-    if (m_handle)
+    if (contentType_AsDefinedInScraper == CONTENT_NONE)
+      return;
+
+    CFileItemList nonRecursiveRemoteItems;
+    CDirectory::GetDirectory(strDirectory, nonRecursiveRemoteItems, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(), DIR_FLAG_DEFAULTS);
+
+    for (const auto& remoteItem : nonRecursiveRemoteItems)
     {
-      m_handle->SetText(g_localizeStrings.Get(20415));
-    }
+      // Support scan cancellation
+      if (m_bStop) {
+        break;
+      }
 
-    /*
-     * Remove this path from the list we're processing. This must be done prior to
-     * the check for file or folder exclusion to prevent an infinite while loop
-     * in Process().
-     */
-    std::set<std::string>::iterator it = m_pathsToScan.find(strDirectory);
-    if (it != m_pathsToScan.end())
-      m_pathsToScan.erase(it);
+      if (m_handle) {
+        m_handle->SetText(remoteItem->GetMovieName(true));
+      }
 
-    // load subfolder
-    CFileItemList items;
-    bool foundDirectly = false;
-    bool bSkip = false;
+      // For TV shows we store only the folders
+      if(contentType_AsDefinedInScraper == CONTENT_TVSHOWS) {
+        if (remoteItem->m_bIsFolder) {
+          remoteDirectoryItems.Add(remoteItem);
 
-    SScanSettings settings;
-    ScraperPtr info = m_database.GetScraperForPath(strDirectory, settings, foundDirectly);
-    CONTENT_TYPE content = info ? info->Content() : CONTENT_NONE;
+          std::string remoteItemPath = remoteItem->GetPath();
+          loadRemoteItems(remoteItemPath, remoteDirectoryItems, rootPathScraperScanSettings, contentType_AsDefinedInScraper);
+        }
+      } else { // For other media types, it's the reverse: we only store files
+        if (remoteItem->m_bIsFolder)
+        {
+          if(rootPathScraperScanSettings.recurse > 0 ) {
+            if( isPathAllowedForScan(strDirectory, rootPathScraperScanSettings, contentType_AsDefinedInScraper) ) {
```

Comment:
coding guidelines: blank after`if`, no blank after `(` please (maybe this is just a typo)